solidusio/solidus_paypal_braintree
# Prepare for 2.0, the last release of this deprecated extension

## Summary

This PR will prepare the extension for 2.0 release. 

2.0 marks the last release (and changes) we will make to this extension. In fact, 2.0 will only be an empty shell that depends on the new extension, called just `solidus_braintree`. 

To facilitate the migration, when upgrading from 1.2 to 2.0, users will see a deprecation warning to show them what to do:

```
❯ gem install solidus_paypal_braintree-2.0.0.gem

ACTION REQUIRED: This extension has been renamed to solidus_braintree.

To upgrade to the new name, follow the instructions here:
https://github.com/solidusio/solidus_braintree/wiki/Upgrading-from-SolidusPaypalBraintree-To-SolidusBraintree

Successfully installed solidus_paypal_braintree-2.0.0
1 gem installed
```

This release has breaking changes, that's why it's a major bump. But, following the instructions provided in the [SolidusBraintree Wiki](https://github.com/solidusio/solidus_braintree/wiki/Upgrading-from-SolidusPaypalBraintree-To-SolidusBraintree), the upgrade to the new name will be pretty smooth. Of course, there might be stores with heavy customizations where this won't be a minute.

## Testing

To test this PR (at least, this is what I did locally):

1. create a new store with the `solidus_frontend` frontend
2. install `solidus_paypal_braintree` 1.2
3. upgrade `solidus_paypal_braintree` to 2.0
4. the deprecation warning is shown while installing the new gem's version
5. upgrading the store with the instructions provided, everything works as before.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.
